### PR TITLE
elasticsearch: use backpressure to resolve streaming request flakes

### LINKED ISSUE #3197: Flakey Test: ITElasticsearchHealthCheck.oneHealthy
```
[ERROR] zipkin2.server.internal.elasticsearch.ITElasticsearchHealthCheck.oneHealthy  Time elapsed: 0.078 s  <<< FAILURE!

org.opentest4j.AssertionFailedError: 

Expecting:

 <false>

to be equal to:

 <true>

but was not.

	at zipkin2.server.internal.elasticsearch.ITElasticsearchHealthCheck.oneHealthy(ITElasticsearchHealthCheck.java:106)
```

### PR DESCRIPTION
**Pull request by minwoox**
The `HttpResponse` was sent before the request are fully sent.
So the request was aborted after getting the response. This changes
to send the response after the request is fully received.

Fixes #3197

### CODE REVIEW

**Comment by minwoox** [comment]
CI failed with the same cause in another test. 😅  Let me fix that too, after lunch. 😄

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java [hunk 1]
```diff
@@ -299,7 +301,10 @@ CheckResult ensureIndexTemplatesAndClusterReady(String index) {
     try {
       version(); // ensure the version is available (even if we already cached it)
       ensureIndexTemplates(); // called only once, so we have to double-check health
-      AggregatedHttpRequest request = AggregatedHttpRequest.of(GET, "/_cluster/health/" + index);
+      final RequestHeaders requestHeaders = RequestHeaders.builder(GET, "/_cluster/health/" + index)
```
Comment (codefromthecrypt) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java]: maybe add a comment on why `endOfStream(true)`?

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java]: Not sure this is solved with this change. Also, I think I have to fix the code in Armeria. 😅 
Let's rebuild a couple of times more.

Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java]: travis will run out of breath :)

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java]: Oops, I got this error.
https://travis-ci.org/github/openzipkin/zipkin/builds/735943469#L7329

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java]: Let me revisit this later after I finished my training session. 😅

**Comment by minwoox** [comment]
Finally, it succeeded. 😅 
Let me rebuild a couple of times though.

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java [hunk 2]
```diff
@@ -205,10 +205,15 @@ public Factory(WebClient httpClient) {
     final HttpResponse response;
     try (SafeCloseable ignored =
            Clients.withContextCustomizer(ctx -> ctx.logBuilder().name(name))) {
-      HttpRequestWriter httpRequest = HttpRequest.streaming(request.headers());
-      response = httpClient.execute(httpRequest);
-      request.writeBody(httpRequest::tryWrite);
-      httpRequest.close();
+      final RequestHeaders headers = request.headers();
```
Comment (codefromthecrypt) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: maybe make a comment as this looks like you have to know that isEndOfStream means a body was already written (it seems like explicit lifecycle control of HTTP/2 stream)

Ex.
```suggestion
      // ElasticsearchStorage healthCheck sets endOfStream to ensure it gets an HTTP response synchronously.
      // The below ensures that flag isn't ignored.
      final RequestHeaders headers = request.headers();
```

File: zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/ITElasticsearchHealthCheck.java [hunk 3]
```diff
@@ -53,19 +57,31 @@
...
     @Override protected void configure(ServerBuilder sb) {
-      sb.service("/", (ctx, req) -> VERSION_RESPONSE.toHttpResponse());
+      sb.service("/", (ctx, req) -> sendResponseAfterAggregate(req, VERSION_RESPONSE));
       sb.service("/_cluster/health", HealthCheckService.of(server1Health));
-      sb.serviceUnder("/_cluster/health/", (ctx, req) -> GREEN_RESPONSE.toHttpResponse());
+      sb.serviceUnder("/_cluster/health/", (ctx, req) -> sendResponseAfterAggregate(req, GREEN_RESPONSE));
     }
   };
 
+  private static HttpResponse sendResponseAfterAggregate(HttpRequest req, AggregatedHttpResponse response) {
```
Comment (codefromthecrypt) [inline zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/ITElasticsearchHealthCheck.java]:
```suggestion
  /** This ensures the response is sent after the request is fully read?? */
  private static HttpResponse sendResponseAfterAggregate(HttpRequest req, AggregatedHttpResponse response) {
```

**Review by codefromthecrypt — APPROVED** [review]
Thanks I like the green color a lot!

Can you help with comments to ensure I understand?

Also, is it temporary that we have to check the isEndOfStream flag? EG, could armeria do that internally? If so, possibly mark an issue in a comment so we don't forget?

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java (hunk 2 above)
Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: I realized that we might not need this change. 😅  So I revert it.

File: zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/ITElasticsearchHealthCheck.java (hunk 3 above)
Comment (minwoox) [reply zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/ITElasticsearchHealthCheck.java]: Added.

**Comment by minwoox** [comment]
@adriancole Do you have a chance to rebuild this a couple of times? 😅

**Comment by codefromthecrypt** [comment]
I clicked rebuild.. you can also rebase into one commit off latest master and force push your branch. pretty confident so far

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java [hunk 4]
```diff
@@ -206,9 +208,9 @@ public Factory(WebClient httpClient) {
     try (SafeCloseable ignored =
            Clients.withContextCustomizer(ctx -> ctx.logBuilder().name(name))) {
       HttpRequestWriter httpRequest = HttpRequest.streaming(request.headers());
-      response = httpClient.execute(httpRequest);
       request.writeBody(httpRequest::tryWrite);
```
Comment (codefromthecrypt) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: I noticed we aren't defensive here like we are on line 109 about empty content. should we be?

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: In order to do that, I have to add a method or change the `writeBody` method in `RequestSupplier`.
So I thought just checking it inside was okay, but if you think we need to defense there I can add it. 😄

Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: In general I'm worried about if we have to be defensive external to the armeria or not, based on line 109. It is scary that we have to check first..

Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: Nevermind. this is our own code.. it is still curious. I'll add a comment suggestion..

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java [hunk 5]
```diff
@@ -107,7 +107,9 @@
     }
 
     @Override public void writeBody(RequestStream requestStream) {
-      requestStream.tryWrite(request.content());
+      if (!request.content().isEmpty()) {
```
Comment (codefromthecrypt) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]:
```suggestion
      // Not all HttpCalls include content. Only try to write the stream when there is.
      if (!request.content().isEmpty()) {
```

File: benchmarks/pom.xml [hunk 6]
```diff
@@ -45,6 +45,13 @@
       <groupId>com.squareup.moshi</groupId>
       <artifactId>moshi</artifactId>
       <version>1.11.0</version>
+      <exclusions>
```
Comment (codefromthecrypt) [inline benchmarks/pom.xml]: did this break something or was it merge drift?

Comment (minwoox) [reply benchmarks/pom.xml]: https://github.com/openzipkin/zipkin/pull/3236#discussion_r505201557
yeah, it breaks the build.

Comment (codefromthecrypt) [reply benchmarks/pom.xml]: thx!

**Comment by codefromthecrypt** [comment]
OK will merge on last green. the comment editing should show this is all working :D Massive thanks @minwoox!

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java [hunk 7]
```diff
@@ -206,9 +209,9 @@ public Factory(WebClient httpClient) {
     try (SafeCloseable ignored =
            Clients.withContextCustomizer(ctx -> ctx.logBuilder().name(name))) {
       HttpRequestWriter httpRequest = HttpRequest.streaming(request.headers());
-      response = httpClient.execute(httpRequest);
       request.writeBody(httpRequest::tryWrite);
       httpRequest.close();
+      response = httpClient.execute(httpRequest);
```
Comment (anuraaga) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: Is this change needed? I think the previous pattern was to allow streaming the request to Elasticsearch while serializing spans. Now, this is effectively `AggregatedHttpRequest` from what I understand, which we could switch to instead, but it would be better if we can use the streaming pattern here since this may be a significant performance regression.

Comment (anuraaga) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: Of course stability is more important - just confirming whether this was actually required though.

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: Yeah, the problem was that the request is aborted if the client receives the response before the request is fully sent. How about using `whenConsumed()` then?

Comment (anuraaga) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: But didn't you change it so the test doesn't do that anymore? Is there a problem in the business logic here with the previous pattern, I thought it's the idiomatic way of streaming a request in Armeria isn't it?

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: >  I thought it's the idiomatic way of streaming a request in Armeria isn't it?

For the streaming request, yeah I think so. It's just that the request is aborted while streaming if the client receives the corresponding response.

Previously, we call the `writeBody` after sending the request headers which means that the streaming body might be sent one by one.
But with this change right now, yeah, we buffer all the stream body into the queue first and then send them all.

If we can use `whenConsumed()` we can accomplish the same way in which the streaming body is sent one by one exactly.

Comment (anuraaga) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: Hmm - not sure why `whenConsumed` is required for streaming the request since it seems to complicate it, but if it's the idiomatic way than that sounds fine to me, can we update to use it?

**Comment by codefromthecrypt** [comment]
if you don't mind, rebase and squash onto latest master. not a big deal if can't

**Comment by minwoox** [comment]
> if you don't mind, rebase and squash onto latest master. not a big deal if can't

After I merged the master, I got this error while building
`Some files do not have the expected license header`
How can I check which files do not have the expected license?

**Comment by codefromthecrypt** [comment]
in `.github/CONTRIBUTING.md` there's this `./mvnw com.mycila:license-maven-plugin:format`

I corrected the problematic license in master, but you can run it again after rebase anyway.

**Comment by minwoox** [comment]
Let me merge master again, then. 😄

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java (hunk 7 above)
Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: @anuraaga Changed it. PTAL.

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java [hunk 8]
```diff
@@ -148,13 +151,23 @@ public BulkCallBuilder(ElasticsearchStorage es, float esVersion, String tag) {
...
+      writeEntry(writer, 0);
+      return writer;
+    }
+
+    private void writeEntry(HttpRequestWriter writer, int index) {
+      if (index == entries.size()) {
+        writer.close();
+        return;
+      }
+      if (!writer.tryWrite(HttpData.wrap(serialize(alloc, entries.get(index), shouldAddType)))) {
```
Comment (anuraaga) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: This looks fine but still curious why we need to use whenConsumed even though we have this boolean check here? I guess it's effectively respecting backpressure which is a good thing, but surprised it affects the stability.

**Review by anuraaga — APPROVED** [review]

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java (hunk 8 above)
Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: > but surprised it affects the stability.

This doesn't affect the stability, it's just for not buffering all the entries in advance before sending the request.
I think all the flaky tests were about `AggregatedRequestSupplier`, not `BulkCallBuilder`.

Previously, we are doing:
```java
HttpRequestWriter httpRequest = HttpRequest.streaming(request.headers());
// Send the request header to the server before we call httpRequest.close().
response = httpClient.execute(httpRequest);
request.writeBody(httpRequest::tryWrite);
httpRequest.close();
```
There's a high chance that the response is received before the request is complete.
So I changed to:
```java
HttpRequestWriter httpRequest = HttpRequest.streaming(request.headers());
request.writeBody(httpRequest::tryWrite);
httpRequest.close();
// Send the request header and body together to the server.
response = httpClient.execute(httpRequest);
```
which there's less chance that the response is received before the request is complete.

But as you noticed, this can be a problem for `BulkCallBuilder` when it's sending streaming requests.
So I changed to use backpressure for that so it's not buffering at all.

Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: A discussion like this begs for a comment, ya?

Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: Maybe this?

// There's a high chance that the response is received before the request is complete.
// This can be a problem for BulkCallBuilder when it's sending streaming requests.
// Hence, we use backpressure, so it's not buffering at all.

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: but let me make the build pass first. 😅

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java [hunk 9]
```diff
@@ -148,13 +151,24 @@ public BulkCallBuilder(ElasticsearchStorage es, float esVersion, String tag) {
...
-          return;
-        }
+    @Override
+    public HttpRequest get() {
+      final HttpRequestWriter writer = HttpRequest.streaming(headers);
+      writeEntry(writer, 0);
+      return writer;
+    }
+
+    // Use backpressure, so it's not buffering at all.
```
Comment (codefromthecrypt) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java]: Most people will barely understand this, so offering the long form comment directly :)

```suggestion
    // There's a high chance that the response is received before the request is complete.
    // This can be a problem for BulkCallBuilder when it's sending streaming requests.
    // Hence, we use backpressure, so it's not buffering at all.
```

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java]: @adriancole After I changed it, it's not working because there's a still chance that the response is received before the request is fully sent. 😅 
Let me just roll back and fix it in Armeria. Sorry about it. 😅

Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java]: ok we can keep this open as it sounds like even if we are seeing this in tests, it could happen in real life. If the comment drifts later, then we update it along with the armeria release.

We were planning to do a release in a few days, but I think this is a good reason to wait a week if it takes that to sort it upstream.

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java (hunk 8 above)
Comment (anuraaga) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: Thanks for working on this - as an aside, if there isn't one already it'd probably be great to have a documentation section on armeria.dev on how to do a streaming HTTP request with the details we've discussed here :)

Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: Yeah, we need that. 😆 
https://github.com/line/armeria/issues/3119

**Comment by minwoox** [comment]
Let me fix this https://github.com/openzipkin/zipkin/pull/3239 in different PR.
Somehow it's not working. 😓

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java (hunk 8 above)
Comment (minwoox) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: The build passes. I didn't revert this change because I like this style but if you want the older version then I can revert it. 😄

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java [hunk 10]
```diff
@@ -148,13 +151,26 @@ public BulkCallBuilder(ElasticsearchStorage es, float esVersion, String tag) {
...
+    @Override
+    public HttpRequest get() {
+      final HttpRequestWriter writer = HttpRequest.streaming(headers);
+      writeEntry(writer, 0);
+      return writer;
+    }
+
+    // There's a high chance that the response is received before the request is complete.
+    // This can be a problem for BulkCallBuilder when it's sending streaming requests.
+    // Hence, we use backpressure, so it's not buffering at all.
```
Comment (minwoox) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java]: This comment is not true anymore so let me remove it.

**Review by codefromthecrypt — COMMENTED** [review]
Exciting! If you have some time, lets get all 1.2 related in the same bucket?

wanna remove some of the log hushing in `zipkin-server-shared.yml` now?
  * (not sure which are needed anymore)

and maybe the TRACE workaround in `ZipkinHttpConfiguration`

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java [hunk 11]
```diff
@@ -20,10 +20,13 @@
 import com.linecorp.armeria.common.HttpData;
 import com.linecorp.armeria.common.HttpHeaderNames;
 import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpRequestWriter;
 import com.linecorp.armeria.common.MediaType;
 import com.linecorp.armeria.common.RequestContext;
 import com.linecorp.armeria.common.RequestHeaders;
 import com.linecorp.armeria.common.util.Exceptions;
+import com.linecorp.armeria.internal.shaded.guava.collect.ImmutableList;
```
Comment (codefromthecrypt) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java]: revert this? it isn't important to use an immutable list for an internal type and also we don't want a shaded dep here

**Comment by minwoox** [comment]
> and maybe the TRACE workaround in ZipkinHttpConfiguration

I left a comment on this. 😄  https://github.com/openzipkin/zipkin/pull/3236#issuecomment-714185794
Let me solve it in different PR.

**Review by anuraaga — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
```
[INFO] -------------------------------------------------------

[INFO] Running zipkin2.elasticsearch.integration.ITElasticsearchStorageV6

2020-10-22 08:05:47:143 [main] WARN org.testcontainers.utility.RegistryAuthLocator - Failure when attempting to lookup auth config (dockerImageName: testcontainers/ryuk:0.3.0, configFile: /home/travis/.docker/config.json. Falling back to docker-java default behaviour. Exception message: /home/travis/.docker/config.json (No such file or directory)

2020-10-22 08:06:14:605 [main] WARN zipkin2.elasticsearch.integration.ITElasticsearchStorage - The ElasticSearch instance used during IT's is using deprecated features or configuration. This is likely nothing to be really worried about (for example 'xpack.monitoring.enabled' setting), but nevertheless it should be looked at to see if our docker image used during integration tests needs updating for the next version of ElasticSearch. See https://www.elastic.co/guide/en/elasticsearch/reference/current/migration-api-deprecation.htmlfor more information. This is the deprecation warning we received:

{"error":"Incorrect HTTP method for uri [/_migration/deprecations] and method [GET], allowed: [POST]","status":405}

[INFO] Running zipkin2.elasticsearch.integration.ITElasticsearchStorage$ITDependenciesHeavy
```

File: zipkin-storage/elasticsearch/src/test/java/zipkin2/elasticsearch/internal/client/HttpCallTest.java [hunk 12]
```diff
@@ -252,13 +253,15 @@
     server.enqueue(SUCCESS_RESPONSE);
 
     HttpCall.RequestSupplier supplier = new HttpCall.RequestSupplier() {
+
+      private final RequestHeaders headers = RequestHeaders.of(HttpMethod.POST, "/");
```
Comment (codefromthecrypt) [inline zipkin-storage/elasticsearch/src/test/java/zipkin2/elasticsearch/internal/client/HttpCallTest.java]: could this be the cause of the test fail?

Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/test/java/zipkin2/elasticsearch/internal/client/HttpCallTest.java]: nevermind this is a test class..

**Comment by darkmastermindz** [comment]
If I'm reading this correctly, #3557 PR - Has a dependency on this to fix #3197 @minwoox @jcchavezs

**Comment by codefromthecrypt** [comment]
currently looking at renovating this one

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java (hunk 1 above)
Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java]: this comment is out-of date, so closing

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java (hunk 7 above)
Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: I can see that whenConsumed is now used to stream the request, so closing this comment

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java (hunk 8 above)
Comment (codefromthecrypt) [reply zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java:174]: changed the comments per above I hope.

**Comment by codefromthecrypt** [comment]
ok 🤞 salvaging a three year old change!

File: zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/ITElasticsearchHealthCheck.java [hunk 13]
```diff
@@ -47,26 +50,36 @@ public class ITElasticsearchHealthCheck {
 
   static final SettableHealthChecker server1Health = new SettableHealthChecker(true);
 
-  static {
-    // Gives better context when there's an exception such as AbortedStreamException
-    System.setProperty("com.linecorp.armeria.verboseExceptions", "always");
```
Comment (codefromthecrypt) [inline zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/ITElasticsearchHealthCheck.java:52]: already in log config, but also we weren't getting exceptions ;)

**Review by anuraaga — APPROVED** [review]
